### ISSUE TITLE
Output

### ISSUE THREAD

**Issue by sai1e** [ProgrammingHero1/secret_capture#4]
when the mp4 output is stored in the folder

i can't open it
it is showing unsupported format

**Comment by ghost** [comment]
Check the extension in code 
file_name = f'{time_stamp}.mp4

**Comment by sai1e** [comment]
not working

**Comment by ghost** [comment]
Show your code to me.I can help you

**Comment by sai1e** [comment]
here is my code


import datetime
from PIL import ImageGrab
import numpy as np
import cv2
from win32api import GetSystemMetrics

width = GetSystemMetrics(0)
height = GetSystemMetrics(1)
print(width)
print(height)
time_stamp = datetime.datetime.now().strftime('%Y-%m-%d %H-%M-%S')
afile_name = f'{time_stamp}.mp4'
fourcc = cv2.VideoWriter_fourcc('m', 'p', '4', 'v')
captured_video = cv2.VideoWriter('output.mp4', fourcc, 20.0, (width, height))      #file_name

webcam = cv2.VideoCapture(0)

while True:
    img = ImageGrab.grab(bbox=(0, 0,width, height ))
    img_np = np.array(img)
    img_final = cv2.cvtColor(img_np, cv2.COLOR_BGR2RGB)
    cv2.imshow('Secret Capture', img_final)

    captured_video.write(img_final)
    if cv2.waitKey(1) == ord('q'):
        break

**Comment by sai1e** [comment]
Here is my code:

import datetime
from PIL import ImageGrab
import numpy as np
import cv2
from win32api import GetSystemMetrics

width = GetSystemMetrics(0)
height = GetSystemMetrics(1)
print(width)
print(height)
time_stamp = datetime.datetime.now().strftime('%Y-%m-%d %H-%M-%S')
afile_name = f'{time_stamp}.mp4'
fourcc = cv2.VideoWriter_fourcc('m', 'p', '4', 'v')
captured_video = cv2.VideoWriter('output.mp4', fourcc, 20.0, (width, height))      #file_name

webcam = cv2.VideoCapture(0)

while True:
    img = ImageGrab.grab(bbox=(0, 0,width, height ))
    img_np = np.array(img)
    img_final = cv2.cvtColor(img_np, cv2.COLOR_BGR2RGB)
    cv2.imshow('Secret Capture', img_final)
    captured_video.write(img_final)
    if cv2.waitKey(1) == ord('q'):
        break

**Comment by ghost** [comment]
ok I got it , I think you are ending program by using IDE such as vs code , pycharm, sublime etc.
dont to do this just simple press 'q' to end.it will save video file by proper finishing.If this working dont forget to reply. And don't forget to visit my repositories her
https://github.com/bhavyamverma2710 and follow me []()https://github.com/bhavyamverma2710

**Comment by sai1e** [comment]
still not working

**Comment by ghost** [comment]
Brother just press q to end

**Comment by sai1e** [comment]
yes bro
but the same thing is repeating 
the recorded video is not comming

**Comment by bhavyamverma2710** [comment]
> Here is my code:
> 
> import datetime
> from PIL import ImageGrab
> import numpy as np
> import cv2
> from win32api import GetSystemMetrics
> 
> width = GetSystemMetrics(0)
> height = GetSystemMetrics(1)
> print(width)
> print(height)
> time_stamp = datetime.datetime.now().strftime('%Y-%m-%d %H-%M-%S')
> afile_name = f'{time_stamp}.mp4'
> fourcc = cv2.VideoWriter_fourcc('m', 'p', '4', 'v')
> captured_video = cv2.VideoWriter('output.mp4', fourcc, 20.0, (width, height))      #file_name
> 
> webcam = cv2.VideoCapture(0)
> 
> while True:
>     img = ImageGrab.grab(bbox=(0, 0,width, height ))
>     img_np = np.array(img)
>     img_final = cv2.cvtColor(img_np, cv2.COLOR_BGR2RGB)
>     cv2.imshow('Secret Capture', img_final)
>     captured_video.write(img_final)
>     if cv2.waitKey(1) == ord('q'):
>         break
> 

    Ok I will make video on my youtube channel just visit my channel 'bhavyam verma'

**Comment by ghost** [comment]
Yeah just hide it or minimize it the recording is going on, and whenever you want to end it come to that repeating window and press q
